### DISCUSSION TITLE
Mapping fingerprint back to molecules

### DISCUSSION THREAD

**Discussion by binary-alkemi** [rdkit/rdkit#6159] [Q&A] [answered]
Hi all,

I am building classification models using MorganFingerprints (`AllChem.GetMorganFingerprintAsBitVect(mol, 2, nBits=2048)`).

I applied to SHAP to see if certain fingerprint indices are important. 

As, I am exploring ML methods, I would like to know, if it is useful to understand SHAP values of fingerprint indices? If yes, is it possible to backmap these indices back to chemical structures used to build model?

Thanks in advance,
Mandar Kulkarni

**Comment by greglandrum** [comment, marked as answer]
This is covered in the documentation:
https://www.rdkit.org/docs/GettingStartedInPython.html#explaining-bits-from-morgan-fingerprints
and in a recent blog post if you want to use a more modern interface for fingerprint generation:
https://greglandrum.github.io/rdkit-blog/posts/2023-01-18-fingerprint-generator-tutorial.html

**Reply by binary-alkemi** [reply]
Thanks for the answer.
